plugins/BEdita/API/src/Controller/Component/JsonApiComponent.php

```diff
@@ -75,23 +76,26 @@ public function initialize(array $config)
      * Input data parser for JSON API format.
      *
      * @param string $json JSON string.
-     * @return array
+     * @return array JSON API input data array
+     * @throws \Cake\Network\Exception\BadRequestException When the request is malformed
      */
     public function parseInput($json)
     {
+        if (empty($json)) {
+            return [];
+        }
         try {
             $json = json_decode($json, true);
             if (json_last_error() || !is_array($json) || !isset($json['data'])) {
-                throw new \InvalidArgumentException('Invalid JSON');
+                throw new BadRequestException(__d('bedita', 'Invalid JSON input'));
             }
 
             return JsonApi::parseData((array)$json['data']);
         } catch (\InvalidArgumentException $e) {
-            if (Configure::read('debug') && !(empty($json))) {
-                throw $e;
-            }
-
-            return [];
+            throw new BadRequestException([
+                'title' => __d('bedita', 'Bad JSON input'),
```

Comment:
Scrutinizer complains about this part:

https://scrutinizer-ci.com/g/bedita/bedita/inspections/47895a8d-f5cf-4ce2-ae68-8b2d53756a43/issues/files/plugins/BEdita/API/src/Controller/Component/JsonApiComponent.php?status=new&orderField=path&order=asc&honorSelectedPaths=0